test_runner/performance/test_physical_replication.py

```diff
@@ -0,0 +1,232 @@
+import subprocess
+import time
+from pathlib import Path
+from typing import Any, List, Optional
+
+import psycopg2
+import psycopg2.extras
+import pytest
+from fixtures.common_types import Lsn
+from fixtures.log_helper import log
+from fixtures.neon_fixtures import PgBin
+from fixtures.pg_version import PgVersion
+
+from performance.neon_api import (
+    neon_create_endpoint,
+    neon_create_project,
+    neon_delete_project,
+    neon_get_connection_uri,
+    neon_suspend_endpoint,
+    neon_wait_for_operation_to_finish,
+)
+
+
+# Granularity of ~0.5 sec
+def measure_replication_lag(master, replica):
+    start = time.time()
+    master.execute("SELECT pg_current_wal_flush_lsn()")
+    master_lsn = Lsn(master.fetchall()[0][0])
+    while True:
```

Comment:
This could stuck, let's add some timeout/deadline

Reply:
Not sure what's a good value for timeout here either, @kelvich what do you think? What kind of replica lag is considered a bug/it being stuck? Is it possible for a read replica to be 10+ minutes behind?

Reply:
Added 10 min timeout